pkg/registrar/transitgateway.go

```diff
@@ -771,6 +775,36 @@ func (r *TransitGateway) removeFromPrefixList(ctx context.Context, awsCluster *c
 	return nil
 }
 
+func (r *TransitGateway) getTGWGAttachmentSubnetsOrDefault(ctx context.Context, awsCluster *capa.AWSCluster) ([]string, error) {
+	result := make([]string, 0)
+	output, err := r.transitGatewayClient.DescribeSubnets(ctx, &ec2.DescribeSubnetsInput{
+		Filters: []types.Filter{
+			{Name: aws.String("tag:" + capa.NameKubernetesAWSCloudProviderPrefix + awsCluster.Name), Values: []string{"owned", "shared"}},
+			{Name: aws.String("tag:" + subnetTGWAttachementslabel), Values: []string{"true"}},
+			{Name: aws.String("tag:" + subnetRoleLabel), Values: []string{"private"}},
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if output == nil || len(output.Subnets) == 0 {
+		for _, s := range getPrivateSubnetsByAZ(awsCluster.Spec.NetworkSpec.Subnets) {
```

Comment:
This function is used only here and we never really use the map for anything. I'd just make it return the desired slice of subnet IDs.